File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventDetailComponent.java
package org.dhis2.usescases.programEventDetail;

import org.dhis2.commons.di.dagger.PerActivity;
import org.dhis2.usescases.programEventDetail.eventList.EventListComponent;
import org.dhis2.usescases.programEventDetail.eventList.EventListModule;
import org.dhis2.usescases.programEventDetail.eventMap.EventMapComponent;
import org.dhis2.usescases.programEventDetail.eventMap.EventMapModule;

import dagger.Subcomponent;

@PerActivity
@Subcomponent(modules = ProgramEventDetailModule.class)
public interface ProgramEventDetailComponent {
    void inject(ProgramEventDetailActivity activity);

    EventListComponent plus(EventListModule eventListModule);

    EventMapComponent plus(EventMapModule eventMapModule);
}

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventDetailComponent.kt
package org.dhis2.usescases.programEventDetail

import dagger.Subcomponent
import org.dhis2.commons.di.dagger.PerActivity
import org.dhis2.usescases.programEventDetail.eventList.EventListComponent
import org.dhis2.usescases.programEventDetail.eventList.EventListModule
import org.dhis2.usescases.programEventDetail.eventMap.EventMapComponent
import org.dhis2.usescases.programEventDetail.eventMap.EventMapModule

@PerActivity
@Subcomponent(modules = [ProgramEventDetailModule::class])
interface ProgramEventDetailComponent {
    fun inject(activity: ProgramEventDetailActivity)
    operator fun plus(eventListModule: EventListModule): EventListComponent
    operator fun plus(eventMapModule: EventMapModule): EventMapComponent
}
